tests/lib/pkgdb.sh

```diff
@@ -305,19 +305,124 @@ distro_install_build_snapd(){
     fi
 }
 
-# Specify necessary packages which need to be installed on a
-# system to provide a basic build environment for snapd.
-export DISTRO_BUILD_DEPS=()
-case "$SPREAD_SYSTEM" in
-    debian-*|ubuntu-*)
-        DISTRO_BUILD_DEPS=(build-essential curl devscripts expect gdebi-core jq rng-tools git netcat-openbsd)
-        ;;
-    fedora-*)
-        DISTRO_BUILD_DEPS=(mock git expect curl golang rpm-build redhat-lsb-core)
-        ;;
-    opensuse-*)
-        DISTRO_BUILD_DEPS=(osc git expect curl golang-packaging lsb-release netcat-openbsd jq rng-tools)
-        ;;
-    *)
-        ;;
-esac
+pkg_apt_dependencies_generic(){
```

Comment:
I find this naming a bit inconsistent with what we have for fedora and opensuse where there's no reference to rpm, maybe `pkg_dependency_ubuntu_{generic,classic}` would be better.